src/fsharp/SimulatedMSBuildReferenceResolver.fs

```diff
@@ -102,7 +102,7 @@ let internal SimulatedMSBuildResolver =
 
                 try
                     if not found && Path.IsPathRooted r then
-                        if FileSystem.SafeExists r then
+                        if AssemblyReader.Exists (r, bypassFileSystemShim) then
```

Comment:
I'm not sure if this resolver should be updated for the new APIs.
There're few `Directory.Exists` checks that should be updated too if we do it.